src/ruleset.rs

```diff
@@ -551,6 +553,34 @@ pub trait RulesetCreatedAttr: Sized + AsMut<RulesetCreated> + Compatible {
         <Self as AsMut<RulesetCreated>>::as_mut(&mut self).no_new_privs = no_new_privs;
         self
     }
+
+    /// Creates a copy of this object and duplicates the file descriptor with
+    /// CLOEXEC set. The `RulesetRef` object that is produced is intended
+    /// to be a readonly copy of the `RulesetCreated` object, however this
+    /// is not currently supported by the landlock kernel module. The intention
```

Comment:
We should add a warning to make sure nodoby use this new method for security purpose because it would be easy to access the underlying file descriptor and update the related ruleset.

Reply:
RulesetRef is gone now in the latest, I think it makes sense to just move new_ref to the RulesetCreated object itself, no need for a trait. I will also remove the AsRef from the attr trait.